providers/provider-protocol-reference.md

```diff
@@ -0,0 +1,202 @@
+
+For Fiberplane Daemon, these functions are implemented in [Provider Runtime
+crate](https://github.com/fiberplane/fiberplane-rs/blob/main/fiberplane-provider-protocol/fiberplane-provider-runtime/src/spec/mod.rs),
+should you want to inspect the source.
+
+### `make_http_request`
+
+#### Signature
+
+```rust
+pub async fn make_http_request(request: HttpRequest) -> Result<HttpResponse, HttpRequestError>;
+```
+
+#### Purpose
+
+This is the most useful imported method, it allows the providers to make HTTP queries to
+remote hosts through the hosting runtime. A basic example of its usage can be seen in the
+tutorial to create a provider:
+
+```rust
+    let base_url: Url = "https://jsonplaceholder.typicode.com".parse().unwrap();
+    let url = base_url
+        .join("/users")
+        .unwrap();
+
+    let request = HttpRequest {
+        url,
+        headers: None,
+        method: HttpRequestMethod::Get,
+        body: None,
+    };
+
+    let response = make_http_request(request).await?;
+    
+    serde_json::from_slice(&response.body).map_err(|e| Error::Deserialization { message: format!("Could not deserialize payload: {e:?}") })
+```
+
+### `log`
+
+#### Signature
+
+```rust
+pub fn log(message: String);
+```
+
+#### Purpose
+
+This function asks the hosting runtime to log a message. This will make your log
+messages appear in Studio or in the standard output of the Fiberplane Daemon
+instance.
+
+### `random`
+
+#### Signature
+
+```rust
+pub fn random(len: u32) -> Vec<u8>;
+```
+
+#### Purpose
+
+Query the hosting runtime for a source of randomness. Note that there is no
+guarantee that the random values you obtain this way to be cryptographically
+secure: it is entirely dependent on the runtime implementation. You can see the
+implementation of randomness for Fiberplane Daemon in the [Provider Runtime
+crate](https://github.com/fiberplane/fiberplane-rs/blob/6679fbd0f1cbdac7c57422ae699e12bb35bed71b/fiberplane-provider-protocol/fiberplane-provider-runtime/src/spec/mod.rs#L116)
+source code
+
+### `now`
+
+#### Signature
+
+```rust
+pub fn now() -> Timestamp;
+```
+
+#### Purpose
+
+Returns a timestamp of the current time of execution. The usefulness of this
+timestamp comes from what you want to do with timers, or signature
+creation/verification schemes.
+
+
+## Exported functions
+
+It is usually safer to use the Plugin Development Kit macros to let the compiler
+generate correct implementations of these functions
+
+### `get_supported_query_types`
+
+#### Signature
+
+```rust
+pub async fn get_supported_query_types(config: ProviderConfig) -> Vec<SupportedQueryType>;
+```
+
+#### Purpose
+
+Returns the query types supported by this provider.
+This function allows Studio to know upfront which formats will be
+supported, and which providers (and their query types) are eligible to
+be selected for certain use cases.
+
+### `invoke2`
+
+#### Signature
+
+```rust
+pub async fn invoke2(request: ProviderRequest) -> Result<Blob, Error>;
```

Comment:
Maybe we should leave out `, Error`. If people try to copy-paste this into a file where the prelude is imported, it wouldn't work otherwise.